test/stat/stat_test.go

```diff
@@ -0,0 +1,148 @@
+package get
+
+import (
+	"fmt"
+	"os"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/runconduit/conduit/testutil"
+)
+
+//////////////////////
+///   TEST SETUP   ///
+//////////////////////
+
+var TestHelper *testutil.TestHelper
+
+func TestMain(m *testing.M) {
+	TestHelper = testutil.NewTestHelper()
+	os.Exit(m.Run())
+}
+
+type rowStat struct {
+	name       string
+	meshed     string
+	success    string
+	rps        string
+	p50Latency string
+	p95Latency string
+	p99Latency string
+}
+
+//////////////////////
+/// TEST EXECUTION ///
+//////////////////////
+
+// This test retries for up to 20 seconds, since each call to "conduit stat"
+// generates traffic to the controller and prometheus deployments in the conduit
+// namespace, and we're testing that those deployments are properly reporting
+// stats. It's ok if the first few attempts fail due to missing stats, since the
+// requests from those failed attempts will eventually be recorded in the stats
+// that we're requesting, and the test will pass. Note that we're not validating
+// stats for the web and grafana deployments, since we aren't guaranteeing that
+// they're receiving traffic as part of this test.
+func TestCliStatForConduitNamespace(t *testing.T) {
+
+	err := TestHelper.RetryFor(20*time.Second, func() error {
+		out, err := TestHelper.ConduitRun("stat", "deploy", "-n", TestHelper.GetConduitNamespace())
+		if err != nil {
+			t.Fatalf("Unexpected stat error: %v", err)
```

Comment:
if this runs before the control-plane is up, will it error? should it `return err` rather than `t.Fatalf(...` ?

Reply:
Good question -- I could go either way. I think it's safe to assume that, for any of these test subdirectory tests, the root conduit install test has been run first. And if that's the case, then the control plane should definitely be up and available.

My thinking for exiting immediately here on error is that the error indicates that the command exited with a non-0 exit code. If we're spuriously returning non-0 exit codes, I want that to fail the test immediately, rather than retry. The retry is really only for unpopulated metrics, assuming the command completed successfully but didn't quite have the right data.